ami/ml/models/pipeline.py

```diff
@@ -431,11 +447,19 @@ def collect_images(
             skip_processed=skip_processed,
         )
 
+    def choose_backend_for_pipeline(self):
+        # @TODO: Create function to backend from the current project and most recently responded OK to a status check
```

Comment:
Raise an error if no backends are online for this pipeline to run

Reply:
Updated. If no backends are online, then I return an empty pipeline response with an error message that 'No backends are online.' Should the Job status be 'FAILURE'? I noticed that the job status isn't always correct (i.e. when no images are processed it still says 'SUCCESS'. Is the job status handled by the frontend? Or the backend?

Reply:
Raise an Exception instead of returning an empty response with the message.

Also I have made a bunch of improvements to the job status updates in this PR #635, so things might change a little. But still, raising an Exception is the way to go here and should cause the job to fail.

Reply:
![image](https://github.com/user-attachments/assets/9adfa76f-5d80-429a-82c7-426cbb20f35c)

I see, I've updated the error handling in the most [recent commit](https://github.com/RolnickLab/antenna/pull/632/commits/8f72fd333d6eeffb34b38e441dfcea7f6eb1698d). I found that I have to raise an Exception and return an empty response. If I raise just an Exception, the error message "No backends are online" shows up in the job log, but the job status is still SUCCESS.